hardware/plugins/DelayedLink.h

```diff
@@ -253,7 +253,7 @@ namespace Plugins {
 						while (!shared_lib_ && (ent = readdir(lDir)) != NULL)
 						{
 							std::string filename = ent->d_name;
-							if (dirent_is_directory(szLibrary, ent) && (filename.length() > 2))
+							if ((ent->d_type == DT_DIR) && (filename.length() > 2))
```

Comment:
Also, better to compare with "." and ".." because the directory can also be a single or two letter directory other than the dots.

filename.compare(".") != 0 && filename.compare("..") != 0